test/c++/mpi_window.cpp

```diff
@@ -329,91 +328,69 @@ TEST(MPI, WindowRingOneSidedStoreWinAllocSharedSignal) {
 
 TEST(MPI, WindowSharedArray) {
   mpi::communicator world;
-  auto shm           = world.split_shared();
-  int const rank_shm = shm.rank();
+  auto shm = world.split_shared();
 
-  constexpr int const array_size = 23;
+  const int array_size = 23;
 
-  mpi::shared_window<int> win{shm, rank_shm == 0 ? array_size : 0};
+  // Only rank 0 allocates the shared array
+  mpi::shared_window<int> win{shm, shm.rank() == 0 ? array_size : 0};
   std::span array_view{win.base(0), static_cast<std::size_t>(win.size(0))};
 
-  // Fill array with local rank in parallel by chunking the range into the communicator
+  // Fill array in parallel: each rank fills its chunk with array indices
   win.fence();
-  auto slice = itertools::chunk_range(0, static_cast<std::ptrdiff_t>(array_view.size()), shm.size(), shm.rank());
-  for (auto i = slice.first; i < slice.second; ++i) { array_view[i] = static_cast<int>(i); }
+  for (auto i : mpi::chunk(itertools::range(array_size), shm)) { array_view[i] = static_cast<int>(i); }
   win.fence();
 
   // Total sum is just sum of numbers in interval [0, array_size)
-  int sum = std::accumulate(array_view.begin(), array_view.end(), int{0});
+  int sum = std::accumulate(array_view.begin(), array_view.end(), 0);
```

Comment:
Let's be explicit about the initializer in `std::accumulate`. Remember this? https://github.com/TRIQS/nda/pull/25
```suggestion
  int sum = std::accumulate(array_view.begin(), array_view.end(), int{0});
```

Reply:
Sure, we can keep this.